dangerfile.js

```diff
@@ -1,31 +1,37 @@
-// eslint-disable-next-line import/no-extraneous-dependencies, no-unused-vars, object-curly-newline
-import { danger, warn, fail, message } from 'danger';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { danger, warn, fail } from 'danger';
 
-const newChangelog = danger.git.created_files.filter((filePath) => {
-  const srcFilePattern = /^packages\/([a-z-])*\/CHANGELOG.md/i;
-  return srcFilePattern.test(filePath);
-});
+const BIG_PR_THRESHOLD = 1000;
+const CHANGELOG_PATTERN = /^packages\/terra-([a-z-])*\/CHANGELOG\.md/i;
 
-const modifiedChangelog = danger.git.modified_files.filter((filePath) => {
-  const srcFilePattern = /^packages\/([a-z-])*\/CHANGELOG.md/i;
-  return srcFilePattern.test(filePath);
-});
+const changedFiles = danger.git.created_files.concat(danger.git.modified_files);
+
+const changedChangelogs = new Set();
+const changedPackages = new Set();
+
+changedFiles.forEach((file) => {
+  // file isn't in a package so it has no changelog, skip further processing
+  if (!file.includes('packages')) {
+    return;
+  }
+
+  const packageName = file.split('packages/')[1].split('/')[0];
 
-const modifiedSrcFiles = danger.git.modified_files.filter((filePath) => {
-  const srcFilePattern = /^packages\/([a-z-])*\/src/i;
-  return srcFilePattern.test(filePath);
+  if (CHANGELOG_PATTERN.test(file)) {
+    changedChangelogs.add(packageName);
+  } else { // file is in a package and was changed - we need a changelog
```

Comment:
We may want to see what happens if any project specific file ( non-package file ) was change. Anything at the root level here: https://github.com/cerner/terra-framework

Reply:
I've tested that by running `danger local`. This branch doesn't have any changes in a package so it doesn't need a changelog entry. I suppose additional verification would be Travis passing the build.

Reply:
I had forgotten you added that return statement above. That should be safe guarding any root level file changes from making here.